src/main/java/hudson/plugins/git/util/BuildData.java

```diff
@@ -243,4 +243,22 @@ public String toString() {
                 ",buildsByBranchName="+buildsByBranchName+
                 ",lastBuild="+lastBuild+"]";
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
> I thought I had used a pure "git revert", yet that's not what it shows.

Possibly you did just that, and it produced a “merge” conflict, and you then resolved it incorrectly before committing.